lib/widgets/editor/editor_toolbar.dart

```diff
@@ -0,0 +1,504 @@
+import '../../resources/links.dart';
+import '../../url_launcher.dart';
+import '../../util/async_store_listener.dart';
+import '../../util/extensions/api.dart';
+import '../../util/extensions/spaced.dart';
+import '../../util/files.dart';
+import '../../util/mobx_provider.dart';
+import '../../util/observer_consumers.dart';
+import '../../util/text_lines_iterator.dart';
+import 'editor.dart';
+import 'editor_picking_dialog.dart';
+import 'editor_toolbar_store.dart';
+
+class _Reformat {
+  final String text;
+  final int selectionBeginningShift;
+  final int selectionEndingShift;
+  _Reformat({
+    required this.text,
+    this.selectionBeginningShift = 0,
+    this.selectionEndingShift = 0,
+  });
+}
+
+enum HeaderLevel {
+  h1(1),
+  h2(2),
+  h3(3),
+  h4(4),
+  h5(5),
+  h6(6);
+
+  const HeaderLevel(this.value);
+  final int value;
+}
+
+class EditorToolbar extends HookWidget {
+  final EditorController controller;
+
+  static const _height = 50.0;
+
+  const EditorToolbar(this.controller);
+
+  @override
+  Widget build(BuildContext context) {
+    final visible = useListenable(controller.focusNode).hasFocus;
+
+    return MobxProvider(
+      create: (context) => EditorToolbarStore(controller.instanceHost),
+      child: Builder(builder: (context) {
+        return AsyncStoreListener(
+          asyncStore: context.read<EditorToolbarStore>().imageUploadState,
+          child: AnimatedSwitcher(
+            duration: kThemeAnimationDuration,
+            transitionBuilder: (child, animation) {
+              final offsetAnimation =
+                  Tween<Offset>(begin: const Offset(0, 1.5), end: Offset.zero)
+                      .animate(animation);
+
+              return SlideTransition(
+                position: offsetAnimation,
+                child: child,
+              );
+            },
+            child: visible
+                ? Material(
+                    color: Theme.of(context).canvasColor,
+                    child: SafeArea(
+                      child: SizedBox(
+                        height: _height,
+                        width: double.infinity,
+                        child: SingleChildScrollView(
+                          scrollDirection: Axis.horizontal,
+                          child: _ToolbarBody(
+                            controller: controller.textEditingController,
+                            instanceHost: controller.instanceHost,
+                          ),
+                        ),
+                      ),
+                    ),
+                  )
+                : const SizedBox.shrink(),
+          ),
+        );
+      }),
+    );
+  }
+
+  static const safeArea = SizedBox(height: _height);
+}
+
+class BottomSticky extends StatelessWidget {
+  final Widget child;
+  const BottomSticky({required this.child});
+
+  @override
+  Widget build(BuildContext context) => Positioned(
+        bottom: 0,
+        child: SizedBox(
+          width: MediaQuery.of(context).size.width,
```

Comment:
`width: double.infinity` wont suffice?

Reply:
it was causing an error, that was the reason why previously i was reluctant to use Positioned, because i couldn't figure out what was messing up